host-controller/src/main/java/org/jboss/as/host/controller/resources/ServerConfigResourceDefinition.java

```diff
@@ -219,20 +231,34 @@ public void registerCapabilities(ManagementResourceRegistration resourceRegistra
     }
 
     public static void registerServerLifecycleOperations(final ManagementResourceRegistration resourceRegistration, final ServerInventory serverInventory) {
-        final ServerStartHandler startHandler = new ServerStartHandler(serverInventory);
-        resourceRegistration.registerOperationHandler(ServerStartHandler.DEFINITION, startHandler);
-        final ServerRestartHandler restartHandler = new ServerRestartHandler(serverInventory);
-        resourceRegistration.registerOperationHandler(ServerRestartHandler.DEFINITION, restartHandler);
-        final ServerStopHandler stopHandler = new ServerStopHandler(serverInventory);
-        resourceRegistration.registerOperationHandler(ServerStopHandler.DEFINITION, stopHandler);
-        final ServerReloadHandler reloadHandler = new ServerReloadHandler(serverInventory);
-        resourceRegistration.registerOperationHandler(ServerReloadHandler.DEFINITION, reloadHandler);
-        resourceRegistration.registerOperationHandler(ServerSuspendHandler.DEFINITION, new ServerSuspendHandler(serverInventory));
-        resourceRegistration.registerOperationHandler(ServerResumeHandler.DEFINITION, new ServerResumeHandler(serverInventory));
-        ServerProcessHandlers.ServerDestroyHandler destroyHandler = new ServerProcessHandlers.ServerDestroyHandler(serverInventory);
-        resourceRegistration.registerOperationHandler(ServerProcessHandlers.DESTROY_OPERATION, destroyHandler);
-        ServerProcessHandlers.ServerKillHandler killHandler = new ServerProcessHandlers.ServerKillHandler(serverInventory);
-        resourceRegistration.registerOperationHandler(ServerProcessHandlers.KILL_OPERATION, killHandler);
+        // Lifecycle ops are deprecated on the server-config resource but not on server
+        boolean serverConfig = resourceRegistration.getPathAddress().getLastElement().getKey().equals(SERVER_CONFIG);
+        ModelVersion deprecatedVersion = serverConfig ? ModelVersion.create(9) : null;
```

Comment:
Note: if this isn't merged for WF 15, the # will need to be increased.

Reply:
This is really a tiny point, I'm not sure if it should be considerated an issue for this PR, but due to this change together with https://github.com/wildfly/wildfly-core/pull/3257/files#diff-d5d53a445abb957562382b7f5df50d9eL51, https://github.com/wildfly/wildfly-core/pull/3257/files#diff-992bfe231fcd944a7d05053a896995eeL91 and https://github.com/wildfly/wildfly-core/pull/3257/files#diff-f2492d9ce98c7f8c0b2c524dd19c61f1L103, we are changing the deprecated since version of `reload`, `suspend` and `resume` operations under server-config from 7.0.0 to 9.0.0, which is not striclty true. I cannot see any big consequence if we are not going to remove those operations soon though, so I let you to your criteria. The other changes LGTM.

Reply:
@yersan Thanks; I'll clean that up.

Reply:
@yersan Fixed.